common/src/main/java/com/wynntils/models/items/ItemModel.java

```diff
@@ -102,6 +102,15 @@ public <T extends WynnItem> Optional<T> asWynnItem(ItemStack itemStack, Class<T>
         return Optional.of((T) wynnItem);
     }
 
+    public <T> Optional<T> asWynnItemPropery(ItemStack itemStack, Class<T> clazz) {
+        Optional<ItemAnnotation> annotationOpt = ItemHandler.getItemStackAnnotation(itemStack);
+        if (annotationOpt.isEmpty()) return Optional.empty();
+        if (!(annotationOpt.get() instanceof WynnItem wynnItem)) return Optional.empty();
+        if (wynnItem.getClass().isAssignableFrom(clazz)) return Optional.empty();
```

Comment:
Shouldn't this be the other way around? Is this the reason for your crashes :)?

Reply:
Yes, indeed, this was the problem :)